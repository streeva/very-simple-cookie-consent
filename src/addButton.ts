export default (
  parentContainer: Element | null,
  text: string,
  btnClass: string,
  handler: any
) => {
  const btn = document.createElement('button')

  btn.innerText = text

  btn.classList.add('vs-cc-bar__button', btnClass)

  btn.addEventListener('click', handler)

  parentContainer?.appendChild(btn)

  return btn
}
