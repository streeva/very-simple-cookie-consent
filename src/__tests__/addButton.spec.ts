import { expect, test } from 'vitest'
import addButton from '../addButton'

test('addButton', () => {
  const parentContainer = document.createElement('div')
  const btnInnerText = 'Example'
  const btnCssClass = 'btn-example-class'
  const callbackFunction = () => {}

  const btn = addButton(
    parentContainer,
    btnInnerText,
    btnCssClass,
    callbackFunction
  )
  expect(btn.innerText).toEqual(btnInnerText)
  expect(btn.classList.contains(btnCssClass)).toBeTruthy()
  expect(parentContainer.toString()).toMatchSnapshot()
})
