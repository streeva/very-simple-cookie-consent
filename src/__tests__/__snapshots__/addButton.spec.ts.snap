// Vitest Snapshot v1

exports[`addButton 1`] = `"<div><button class=\\"vs-cc-bar__button btn-example-class\\">Example</button></div>"`;
